nixos/modules/services/misc/lms.nix

```diff
@@ -0,0 +1,234 @@
+
+    home = mkOption {
+      type = types.path;
+      default = "/var/lib/lms";
+      description = ''
+        Path to the working directory. Must have write privileges in order to
+        create and modify this directory.
+      '';
+    };
+
+    virtualHost = mkOption {
+      type = types.nullOr types.str;
+      default = null;
+      description = ''
+        Name of the nginx virtualhost to setup as reverse proxy. If null, do
+        not setup any virtualhost.
+      '';
+    };
+
+    listenAddress = mkOption {
+      type = types.str;
+      default = "127.0.0.1";
+      description = ''
+        The address on which to bind LMS.
+      '';
+    };
+
+    port = mkOption {
+      type = types.int;
+      default = 5082;
+      description = ''
+        The port on which LMS will listen to.
+      '';
+    };
+
+    theadCount = mkOption {
+      type = types.int;
+      default = 0;
+      description = ''
+        Number of threads to be used to dispatch http requests. 0 means auto
+        detect.
+      '';
+    };
+
+    listenbrainzApiBaseUrl = mkOption {
+      type = types.nullOr types.str;
+      default = null;
+      example = "https://api.listenbrainz.org";
+      description = ''
+        Listen brainz's root API url. If null, use default url.
+      '';
+    };
+
+    listenbrainzMaxSyncListenCount = mkOption {
+      type = types.int;
+      default = 1000;
+      description = ''
+        How many listens to retrieve when syncing. 0 disables sync.
+      '';
+    };
+
+    listenbrainzSyncPeriod = mkOption {
+      type = types.int;
+      default = 1;
+      description = ''
+        How often to resync listens. 0 disables sync.
+      '';
+    };
+
+    acousticbrainzApiBaseUrl = mkOption {
+      type = types.nullOr types.str;
+      default = null;
+      example = "https://acousticbrainz.org";
+      description = ''
+        Acoustic brainz's root API url. If null, use default url.
+      '';
+    };
+
+    loginThrottlerMaxEntries = mkOption {
+      type = types.int;
+      default = 10000;
+      description = ''
+        Max entries in the login throttler (1 entry per IP address. For IPv6,
+        the whole /64 block is used).
+      '';
+    };
+
+    enableSubsonicApi = mkOption {
+      type = types.bool;
+      default = true;
+      description = ''
+        Enable the Subsonic API.
+      '';
+    };
+
+    coverArtMaxSize = mkOption {
+      type = types.int;
+      default = 10;
+      description = ''
+        Max external cover file size in MBytes.
+      '';
+    };
+
+    coverArtCacheSize = mkOption {
+      type = types.int;
+      default = 30;
+      description = ''
+        Max cover cache size in MBytes.
+      '';
+    };
+
+    coverArtJpegQuality = mkOption {
+      type = types.int;
+      default = 75;
+      description = ''
+        JPEG quality for covers (range is 1-100).
+      '';
+    };
```

Comment:
According to NixOS best practices it would be better to create a single `settings` option than to include each of these options. If still desired the `settings` options could include a type hint in for each of the above values by utilizing `freeformType`.

See https://github.com/NixOS/rfcs/blob/master/rfcs/0042-config-option.md for details, and please do not hesitate if you have any questions, or just need a hand hacking it together. I'm always happy to help.

Reply:
100% agree. The stuff here was actually mostly written 1+ year ago; the new approach is definitely nicer. (It is new right? I don’t recall seeing them in the docs back in early 2020.)

Reply:
Yes! Very much so. :+1: